tests/expressions/tests.py

```diff
@@ -2333,6 +2333,34 @@ def test_resolve_output_field(self):
                 expr = Value(value)
                 self.assertIsInstance(expr.output_field, output_field_type)
 
+    def test_mixed_output_field_with_none_returns_field(self):
+        test_cases = [
+            (Value(3.14159) + Value(None), FloatField),
+            (Value(42) + Value(None), IntegerField),
+            (Value(Decimal("3.14")) + Value(None), DecimalField),
+        ]
+
+        for expression, result in test_cases:
+            with self.subTest(expression):
+                self.assertIsInstance(expression.output_field, result)
+
+    def test_mixed_output_field_with_none_returns_none(self):
+        test_cases = [
+            Value(None),
+            Value("a string") + Value(None),
+            Value(datetime.date(2019, 5, 15)) + Value(None),
+            Value(datetime.time(3, 16)) + Value(None),
+            Value(datetime.datetime(2019, 5, 15)) + Value(None),
+            Value(datetime.timedelta(1)) + Value(None),
+            Value(b"") + Value(None),
+            Value(uuid.uuid4()) + Value(None),
+        ]
```

Comment:
As far as I'm aware these test are unrelated with a patch so we should proceed them separately. Please remove them them from this PR.

Reply:
Removed~